pkg/admission/validator/shoot.go

```diff
@@ -91,10 +95,13 @@ func (s *shoot) Validate(ctx context.Context, new, old client.Object) error {
 	return s.validateShootCreation(ctx, shoot)
 }
 
-func (s *shoot) validateShoot(_ context.Context, shoot *core.Shoot, infraConfig *alicloud.InfrastructureConfig, cpConfig *alicloud.ControlPlaneConfig) error {
-
+func (s *shoot) validateShoot(ctx context.Context, shoot *core.Shoot, infraConfig *alicloud.InfrastructureConfig, cpConfig *alicloud.ControlPlaneConfig) error {
+	natGatewayZones, err := s.getEnhancedNatGatewayAvailableZones(ctx, shoot)
+	if err != nil {
+		return err
+	}
 	// Provider validation
-	if errList := alicloudvalidation.ValidateInfrastructureConfig(infraConfig, shoot.Spec.Networking.Nodes, shoot.Spec.Networking.Pods, shoot.Spec.Networking.Services); len(errList) != 0 {
+	if errList := alicloudvalidation.ValidateInfrastructureConfig(infraConfig, shoot.Spec.Networking.Nodes, shoot.Spec.Networking.Pods, shoot.Spec.Networking.Services, natGatewayZones); len(errList) != 0 {
```

Comment:
Would you just push `shootVPCClient` to function `ValidateInfrastructureConfig` and do some refactoring for `alicloudvalidation.ValidateInfrastructureConfig`? The declaration of alicloudvalidation.ValidateInfrastructureConfig may look like: 
```go
func ValidateInfrastructureConfig(infra *apisalicloud.InfrastructureConfig, shootNetworkConfig *Networking, vpcClient VPCClient)
```